src/server/config_registry.cc

```diff
@@ -25,7 +25,6 @@ auto ConfigRegistry::Set(std::string_view config_name, std::string_view value) -
     return SetResult::READONLY;
 
   auto cb = it->second.cb;
-  lk.unlock();
```

Comment:
I removed this, because even though `flag->ParseFrom` is atomic (protected internally by a mutex), it allows other `CONFIG SET` to interleave the atomic operations after the lock is released
```
proactor 1                           proactor 2

config set something 22                  config set something 33
lk.unlock                                           
                                                        lk.lock 
                                                        lk.unlock
                                                        
flag->ParseFlag                              
                                                        flag->ParseFlag
                                                        
cb here sees a different value 
```

One more issue, that I did not solve here is there is a race between the proactors that read the flag 
and the `cb` executed. This might be problematic, for example it could be the case, that `AUTH` reads an updated `password` from `requirepass` which yet has not been updated via it's cb in `config registry`. In that sense, we `Set` should be atomic, and no other readers should see the new value before `Set` has finished (such that its callbacks get executed)

Reply:
I assume that's impossible to achieve with absl flags